Core/include/Acts/EventData/MultiTrajectory.hpp

```diff
@@ -1510,8 +1523,19 @@ class MultiTrajectory {
   }
 
  private:
+  template <typename T, bool RO = ReadOnly, typename = std::enable_if_t<!RO>>
+  void copyDynamicFrom(IndexType dstIdx, const T& src, IndexType srcIdx) {
```

Comment:
I guess this would avoid having to add the extra getter for the columns keys. 

My thinking was to make the backend do as little work as possible, but maybe that's not needed.

Reply:
I'll still need `dynamicKeys_impl`, because otherwise I can't determine what the source columns are, and since the destination backend won't necessarily know what the source backend is, it wouldn't work.

I can push the loop inside the backend still, but it doesn't seem like this reduces repetition. What do you think @tboldagh?

Reply:
Indeed. Trying to think how we could avoid this vector of IDs though.
Could the copyDynamicFrom take as an argument a function that gets invoked for every decoration?
e.g. `copyDynamicFrom(src, srcIndex, inserter);`
where inserter would be:
`[destIndex, &destContainer](hashId, std::any);`

If you think this is overengeneering then I am also ok.

Reply:
I mean it could be defined to return an iterable range, instead of a concrete vector. That way at least the vector transient backend could generate it on the fly when called.

Reply:
Not sure if it is needed. I.e. the method would be called on the source container which would be responsible to iterate over whatever it suits best and the call the provided as last arg will be fed  with key & value pairs.

Reply:
How about this?

```
TrackState::copyDynamicFrom(..., inserter) {
 m_traj->copyDynamicFrom(index, inserter);
}

VectorMTJ::copyDynamicFrom_impl(srcIndex, inserter) {
  for ( auto [hash, data]: m_dynamic) {
   inserter(hash, data.get(srcIndex));
  }
}
```

Reply:
We had a chat on the side. The above is considered overengeeenered (a very bad thing!).

Reply:
Switched to the concept only requiring a range that gives iterators that yield `HashedString`, without requirement on storage.

The shipped backends now return an iterator adapter that unpacks the keys from `unordered_map` dynamically.